src/util/c_sizeof.cpp

```diff
@@ -23,14 +23,13 @@ expr2tc c_sizeof(const type2tc &src, const namespacet &ns)
     // Right now we just introduce a symbol __ESBMC_INF_SIZE to represent
     // the "infinity" expression in irep1. (As of now migrating
     // the latter into irep2 results in an error.)
-    expr2tc inf_size = symbol2tc(get_uint64_type(), "__ESBMC_INF_SIZE");
-    expr2tc subtype_size = constant_int2tc(
-      get_uint64_type(), type_byte_size(to_array_type(t).subtype));
-    return mul2tc(get_uint64_type(), inf_size, subtype_size);
+    expr2tc subtype_size = type_byte_size_expr(to_array_type(t).subtype, &ns);
+    type2tc sz_type = subtype_size->type;
+    expr2tc inf_size = symbol2tc(sz_type, "__ESBMC_INF_SIZE");
+    return mul2tc(sz_type, inf_size, subtype_size);
```

Comment:
Just using `size_type2()` for `get_uint64_type` does not work?

Reply:
It's size_type2() but in addition allows dynamically sized arrays, such as VLAs, where previously this code would break. That's the fixed knownbug regression.